core/chains/evm/logpoller/log_poller_test.go

```diff
@@ -623,6 +623,50 @@ func TestGetReplayFromBlock(t *testing.T) {
 	assert.Equal(t, requested, fromBlock)
 }
 
+func TestLogPoller_AsyncReplay(t *testing.T) {
+	th := SetupTH(t, 2, 3, 2)
+	// Commit a few blocks
+	for i := 0; i < 3; i++ {
+		th.Client.Commit()
+	}
+	// process logs up through latest
+	th.LogPoller.pollAndSaveLogs(testutils.Context(t), 1)
+	latest, err := th.LogPoller.LatestBlock()
+	require.NoError(t, err)
+	require.Equal(t, int64(4), latest)
+
+	th.LogPoller.ctx, th.LogPoller.cancel = context.WithCancel(testutils.Context(t))
+
+	// Replay() should return as soon as replay signal is received,
+	// so that replay can proceed asynchronously
+	go func() {
+		time.Sleep(100 * time.Microsecond)
+		<-th.LogPoller.replayStart
+	}()
```

Comment:
[time.AfterFunc](https://pkg.go.dev/time#AfterFunc) would drop right in here:
```suggestion
	time.AfterFunc(100 * time.Microsecond, func() {
		<-th.LogPoller.replayStart
	})
```
Especially nice on the others, without the closure boilerplate: `time.AfterFunc(100 * time.Microsecond, cancel)`

Also, this approach returns a timer that can be stopped on cleanup - seems like nbd in this case though.

Reply:
Thanks!